src/lib/Component/Renderer/DefaultRenderer.php

```diff
@@ -21,10 +22,16 @@ final class DefaultRenderer implements RendererInterface
 
     private EventDispatcherInterface $eventDispatcher;
 
-    public function __construct(Registry $registry, EventDispatcherInterface $eventDispatcher)
-    {
+    private TwigComponentCollector $collector;
+
+    public function __construct(
+        Registry $registry,
+        EventDispatcherInterface $eventDispatcher,
+        TwigComponentCollector $collector
+    ) {
         $this->registry = $registry;
         $this->eventDispatcher = $eventDispatcher;
+        $this->collector = $collector;
```

Comment:
Consider using events to collect data, as this would allow detaching this functionality from the `DefaultRenderer` instance (single responsibility pricinple and all that :sweat_smile:).

Alternatively, consider creating `TraceableDefaultRenderer` that would maintain it's calls/data.

See `Symfony\Component\Form\Extension\DataCollector\EventListener\DataCollectorListener` for Symfony forms & event listening-based collection.

See `Symfony\Component\HttpClient\TraceableHttpClient` & `Symfony\Component\HttpClient\DataCollector\HttpClientDataCollector` for Traceable variant.